### PR TITLE
IOS-10010 Add new brand O2New

### PR DESCRIPTION
**Pull request by alejandroruizponce**
## 🎟️ **Jira ticket**
https://jira.tid.es/browse/IOS-10010

## 🥅 **What's the goal?**
Add the new brand called **O2New** with all its new colors, fonts, radius, etc.

More info here: https://github.com/Telefonica/mistica-design/blob/production/tokens/o2-new.json

## 🚧 **How do we do it?**
For this purpose, the following steps have been taken:
- Firstly to be able to carry out this development without the token importer failing due to the fact that currently there are gradients inside the color tokens that are not of color type this PR has been carried out to solve this problem: 
https://github.com/Telefonica/mistica-ios/pull/357
- Makefile file has been updated to capture the values of the json file in O2New's mistica-design when we do a `make skin`.
- Once the Makefile has been updated, using the command `make skin ref=production` we have downloaded and created all the palettes related to colors, radius, fonts and others.
- In MisticaConfig we have added the new brand O2New and in the project assets we have added the same assets that existed for O2.

Also the palettes of the rest of the brands have been updated, it would not have been necessary if the MisticaColors protocol had not changed with new tokens, but unfortunately it has changed and it is not possible that O2New has those new tokens and the rest of the brands do not have them. For that reason we have taken the opportunity to update the rest of the palettes as well.

**More info here:** https://confluence.tid.es/display/CTO/%5BAPPS%5D+Shared+Spec%3A+Mistica+gradients

### CODE REVIEW

File: Sources/MisticaCommon/Colors/O2NewColorPalette.swift [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+    let tagBackgroundInactive = O2NewColors.palette.grey20 | O2NewColors.palette.darkModeGrey6
+
+    let tagBackgroundSuccess = O2NewColors.palette.o2Green10 | O2NewColors.palette.darkModeGrey6
+
+    let tagBackgroundWarning = O2NewColors.palette.o2Orange10 | O2NewColors.palette.darkModeGrey6
+
+    let tagBackgroundError = O2NewColors.palette.o2Red10 | O2NewColors.palette.darkModeGrey6
+
+    #warning("Gradients tokens not formatted to UIColor. Set old O2 color values by default.")
```
Comment (alejandroruizponce) [inline Sources/MisticaCommon/Colors/O2NewColorPalette.swift:258]: Temporal fix. **Gradient tokens that we have avoided importing thanks to the previously mentioned PR (https://github.com/Telefonica/mistica-ios/pull/357) have no value in the O2New palette and therefore we are waiting for Design to establish a definitive approach to know if the gradients will be separated in a separate palette or if they are finally part of the color palette, we will have to redefine these tokens with a new supertype that allows us to control UIColor and Gradient at the same time.**

**More info here:** https://confluence.tid.es/display/CTO/%5BAPPS%5D+Shared+Spec%3A+Mistica+gradients

File: Sources/MisticaCommon/Colors/BlauColorPalette.swift [hunk 2]
```diff
@@ -12,116 +12,252 @@ struct BlauColors: MisticaColors {
...
+
     let textLinkSnackbar = BlauColors.palette.blauPurple30
+
     let control = BlauColors.palette.grey2 | BlauColors.palette.darkModeGrey6
+
     let controlActivated = BlauColors.palette.blauBlueSecondary
-    let controlInverse = BlauColors.palette.white | BlauColors.palette.darkModeGrey6
+
+    let controlInverse = BlauColors.palette.blauBluePrimary30 | BlauColors.palette.darkModeGrey6
+
```
Comment (amegias) [inline Sources/MisticaCommon/Colors/BlauColorPalette.swift:135]: I would revert this, this PR shouldn't affect other OBs

File: Sources/MisticaCommon/Colors/MovistarColorPalette.swift [hunk 3]
```diff
@@ -12,116 +12,252 @@ struct MovistarColors: MisticaColors {
...
+
     let textLinkSnackbar = MovistarColors.palette.movistarBlue30
+
     let control = MovistarColors.palette.grey3 | MovistarColors.palette.darkModeGrey6
+
     let controlActivated = MovistarColors.palette.movistarBlue
-    let controlInverse = MovistarColors.palette.white | MovistarColors.palette.darkModeGrey6
+
+    let controlInverse = MovistarColors.palette.movistarBlue20 | MovistarColors.palette.darkModeGrey6
+
```
Comment (amegias) [inline Sources/MisticaCommon/Colors/MovistarColorPalette.swift:135]: revert

File: Sources/MisticaCommon/Colors/O2ColorPalette.swift [hunk 4]
```diff
@@ -12,116 +12,252 @@ struct O2Colors: MisticaColors {
...
+
     let textLinkSnackbar = O2Colors.palette.o2BluePrimary30
+
     let control = O2Colors.palette.grey3 | O2Colors.palette.darkModeGrey6
+
     let controlActivated = O2Colors.palette.o2BluePrimary | O2Colors.palette.o2BluePrimary30
-    let controlInverse = O2Colors.palette.white | O2Colors.palette.darkModeGrey6
+
+    let controlInverse = O2Colors.palette.o2BluePrimary30 | O2Colors.palette.darkModeGrey6
+
```
Comment (amegias) [inline Sources/MisticaCommon/Colors/O2ColorPalette.swift:135]: revert

File: Sources/MisticaCommon/Colors/TelefonicaColorPalette.swift [hunk 5]
```diff
@@ -12,116 +12,252 @@ struct TelefonicaColors: MisticaColors {
...
     let textLinkDanger = TelefonicaColors.palette.coral
+
     let textLinkSnackbar = TelefonicaColors.palette.telefonicaBlue30
+
     let control = TelefonicaColors.palette.grey3 | TelefonicaColors.palette.darkModeGrey6
+
     let controlActivated = TelefonicaColors.palette.telefonicaBlue
-    let controlInverse = TelefonicaColors.palette.white | TelefonicaColors.palette.darkModeGrey6
+
+    let controlInverse = TelefonicaColors.palette.grey1 | TelefonicaColors.palette.darkModeGrey6
```
Comment (amegias) [inline Sources/MisticaCommon/Colors/TelefonicaColorPalette.swift:134]: revert

File: Sources/MisticaCommon/Colors/VivoColorPalette.swift [hunk 6]
```diff
@@ -12,116 +12,252 @@ struct VivoColors: MisticaColors {
...
+
     let textLinkSnackbar = VivoColors.palette.vivoPurpleLight50
+
     let control = VivoColors.palette.grey3 | VivoColors.palette.darkModeGrey6
+
     let controlActivated = VivoColors.palette.vivoPurple | VivoColors.palette.vivoPurpleLight80
-    let controlInverse = VivoColors.palette.white | VivoColors.palette.darkModeGrey6
+
+    let controlInverse = VivoColors.palette.vivoPurpleLight50 | VivoColors.palette.darkModeGrey6
+
```
Comment (amegias) [inline Sources/MisticaCommon/Colors/VivoColorPalette.swift:135]: revert

File: Sources/MisticaCommon/Colors/VivoNewColorPalette.swift [hunk 7]
```diff
@@ -12,116 +12,252 @@ struct VivoNewColors: MisticaColors {
...
     let textLinkDanger = VivoNewColors.palette.pepper
+
     let textLinkSnackbar = VivoNewColors.palette.vivoPurpleLight20 | VivoNewColors.palette.vivoPurpleLight50
+
     let control = VivoNewColors.palette.grey4 | VivoNewColors.palette.darkModeGrey6
+
     let controlActivated = VivoNewColors.palette.vivoPurple | VivoNewColors.palette.vivoPurpleLight80
-    let controlInverse = VivoNewColors.palette.white | VivoNewColors.palette.darkModeGrey6
+
+    let controlInverse = VivoNewColors.palette.vivoPurpleLight50 | VivoNewColors.palette.darkModeGrey6
```
Comment (amegias) [inline Sources/MisticaCommon/Colors/VivoNewColorPalette.swift:134]: same

**Review by amegias — APPROVED** [review]
Good work

File: Sources/MisticaCommon/Colors/BlauColorPalette.swift (hunk 2 above)
Comment (alejandroruizponce) [reply Sources/MisticaCommon/Colors/BlauColorPalette.swift]: Reverted!

**Review by franciscojrp — APPROVED** [review]

**Review by idenjoe — APPROVED** [review]

**Review by aweell — APPROVED** [review]

File: Tests/MisticaSwiftUITests/UI/GradientTests.swift [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import SwiftUI
+import XCTest
+
+final class GradientTests: XCTestCase {
+    override class func setUp() {
+        isRecording = false
+    }
+    
+    func testGradientInView() {
+        let misticaGradient: MisticaGradient = .init(colors: [.backgroundBrandBottom, .blue, .backgroundBrandSecondary], stops: [0, 0.5, 1], angle: 180)
```
Comment (amegias) [inline Tests/MisticaSwiftUITests/UI/GradientTests.swift:20]: is the angle working?

File: Sources/MisticaCommon/Colors/O2NewColorPalette.swift [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+    public let o2Red65 = UIColor(hex: "#BF4343")!
+    public let grey20 = UIColor(hex: "#F3F3F5")!
+    public let grey30 = UIColor(hex: "#D9D9DD")!
+    public let grey40 = UIColor(hex: "#B4B4BE")!
+    public let grey45 = UIColor(hex: "#8C8C9A")!
+    public let grey60 = UIColor(hex: "#6E6E77")!
+    public let grey80 = UIColor(hex: "#3C3C46")!
+    public let black = UIColor(hex: "#00001E")!
+    public let white = UIColor(hex: "#FFFFFF")!
+    public let darkModeBlack = UIColor(hex: "#121212")!
```
Comment (yceballost) [inline Sources/MisticaCommon/Colors/O2NewColorPalette.swift:315]: This color is from the previous version of dark mode for o2-new. Can you fetch tokens from mistica-design to ensure the last version of colors in dark mode?

Comment (yceballost) [reply Sources/MisticaCommon/Colors/O2NewColorPalette.swift:315]: aw, I was late 😮‍💨

Comment (alejandroruizponce) [reply Sources/MisticaCommon/Colors/O2NewColorPalette.swift:315]: We have not yet generated a version, I take the opportunity to change it in a PR that we have open. Thanks Yayo!

**Pull request merged**